### ISSUE TITLE
prop-types are ignored when render returns a React.createElement

### ISSUE THREAD

**Issue by hallister** [yannickcr/eslint-plugin-react#205]
One of the features of a library I'm developing is allowing the return tag to be of type string, function or ReactElement. In order to make that work reliably I use an anonymous function as my return tag, so the foot print of my render may look like:

```javascript
render() {
    let Tag = React.DOM.div;

    return Tag({
        className: this.props.className,
        ...this.props
    }, this.props.children());
```

In my usecase the Tag can either be a `React.DOM` element or it can be an anonymous function that returns a cloned `ReactElement`. This *completely* breaks prop validation, I get no errors. Right now I have to insert a random return statement (such as `return(<div/>)`) above my actual return statement to see if there are prop errors.

**Comment by hallister** [comment]
This is actually fixable by wrapping the Tag in a `React.createElement` and it will work for any element (including strings) so closing this.

**Comment by hallister** [comment]
Thought this was fixed, but it is not. If the return function of the render is a React.createElement the react/prop-types validation simply won't fire at all:

```javascript
export class Test extends React.Component {
    render() {
        return React.createElement(
            'div',
            this.props,
            this.props.children
        );
    }
}
```
`react/prop-types` won't ever fire if you return a ReactElement this way. I've tested this with several other props as well to no avail. Putting a return (<div/>); above the return will fire all react/prop-types properly.

**Comment by yannickcr** [comment]
It should be fixed in master by #206 but the fix is not released on npm yet.

**Comment by hallister** [comment]
Just bumped to 3.3.2 and still no luck. Here's a gist of a simple React (es6 + experimental) that doesn't throw any prop-type warnings/errors:

https://gist.github.com/hallister/7a450c921a9907a87cd9

**Comment by yannickcr** [comment]
After test, I correctly have all the warnings with your gist (for `dog`, `cat`, `stuff`, `component` and `children`).

Tested with `eslint@1.4.3`, `eslint-plugin-react@3.3.2` and `babel-eslint@4.1.2`

**Comment by hallister** [comment]
Hey thanks for the versions, didn't realize I was running on a fairly dated eslint (whoops!). Problem solved.
